compiler/rustc_codegen_ssa/src/mir/intrinsic.rs

```diff
@@ -249,6 +249,38 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                     }
                 }
             }
+            sym::fadd_algebraic
+            | sym::fsub_algebraic
+            | sym::fmul_algebraic
+            | sym::fdiv_algebraic
+            | sym::frem_algebraic => match float_type_width(arg_tys[0]) {
```

Comment:
cg_clif also needs this. As Cranelift doesn't have fast-math support, you can just add it to https://github.com/rust-lang/rust/blob/d6c46a23ce19e910225abacc33bcca9d0f549148/compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs#L1159 and https://github.com/rust-lang/rust/blob/d6c46a23ce19e910225abacc33bcca9d0f549148/compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs#L1164-L1171